system/system.go

```diff
@@ -131,34 +131,34 @@ func Command(name string, arg ...string) *Cmd {
 type cmdLogger struct {
 	commandName string
 	stream      string
-	*bytes.Buffer
+	buf         *bytes.Buffer
```

Comment:
For any struct which is passed to some function by interface value, it is always the "outermost" struct (the one which embeds others) which receives the call. This is the same as one would typically see in traditional OOP languages like C++, Java and Python, as long as it's called through an interface.

So I don't understand how this can help. To me it looks like this gives exactly the same behavior as before this change. Are you saying that `cmdLogger.Write()` was not called prior to this change, but with this change, it is?

Reply:
That makes a lot more sense, thanks for looking into it. I knew that `cmdLogger.Write()` wasn't being called but I wasn't sure exactly what was going on behind the scenes other than the data was still being written to the buffer through some other method.